tests/torch2/function_hook/test_disable_tracing.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from torch import nn
+from torch.overrides import _get_current_function_mode_stack
+
+from nncf.experimental.torch2.function_hook.graph.build_graph_mode import build_graph
+from nncf.experimental.torch2.function_hook.graph.graph_visualization import to_pydot
+from nncf.experimental.torch2.function_hook.hook_executor_mode import FunctionHookMode
+from nncf.experimental.torch2.function_hook.wrapper import wrap_model
+from nncf.torch import disable_tracing
+from tests.cross_fw.shared.paths import TEST_ROOT
+from tests.torch2.utils import compare_with_reference_file
+
+REF_DIR = TEST_ROOT / "torch2" / "data" / "function_hook" / "disable_tracing"
+
+
+class ModelNoTrace(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x):
+        x = x + 1
+        return self.foo(x)
+
+    def foo(self, x):
+        mode = _get_current_function_mode_stack()
+        assert len(mode) == 1
+        assert isinstance(mode[0], FunctionHookMode)
+        assert not mode[0].enabled
+        return x - 1
+
+
+disable_tracing(ModelNoTrace.foo)
+
+
+def test_build_graph_with_disable_tracing(regen_ref_data):
+    model = wrap_model(ModelNoTrace())
```

Comment:
What do you think about updating this test in the following way:
1) You check tracing is enabled
2) You wraps the model with the `disable_tracing`, check the tracing is disabled
3) You check that tracing is once again enabled without the `disable_tracing` context

Reply:
> You check tracing is enabled

Tracing is not available for not wrapped model (in user's scenarios of using).
There is no any global tracing context. Instance of FunctionHookMode creates in patched forward https://github.com/openvinotoolkit/nncf/blob/d614c1ed9f2e143895a74d21a75f6369c1decff7/nncf/experimental/torch2/function_hook/wrapper.py#L53

But as i understand you suggestion:
- added test_disable_function_hook_mode that check FunctionHookMode inside and outside disble context.
- add operation after function with disabled tracing